apps/network-alerts/js/attention/notify.js

```diff
@@ -0,0 +1,116 @@
+/* global
+  AudioContext
+ */
+
+(function(exports) {
+  'use strict';
+
+  const SETTINGS = {
+    notificationVolume: 'audio.volume.notification',
+    vibration: 'vibration.enabled'
+  };
+
+  const ATTENTION_PATTERN = [2000, 500, 1000, 500, 1000, 500];
+  ATTENTION_PATTERN.push(...ATTENTION_PATTERN);
+
+  const ATTENTION_SOUND_VOLUME = 0.3;
+
+  function getSetting(key) {
+    if (!navigator.mozSettings) {
+      return Promise.reject(new Error('The mozSettings API is not available.'));
+    }
+
+    return navigator.mozSettings.createLock().get(key).then(
+      result => result[key]
+    );
+  }
+
+  function getSettings(settings) {
+    return Promise.all(
+      settings.map(getSetting)
+    ).catch((e) => {
+      // catch and log errors
+      console.error('Error while retrieving settings', e.message, e);
+      return settings.map(() => null);
+    }).then((results) => {
+      return settings.reduce((result, setting, i) => {
+        result[setting] = results[i];
+        return result;
+      }, {});
+    });
+  }
+
+  // Converts from the ATTENTION_PATTERN (suitable for the Vibration API) to a
+  // Float32Array suitable for the Audio API.
+  function getAttentionCurveWave() {
+    var halfSecDurations = ATTENTION_PATTERN.map(val => val / 500);
+    var result = [];
+    var currentValue = ATTENTION_SOUND_VOLUME;
+
+    halfSecDurations.forEach(duration => {
+      for (var i = 0; i < duration; i++) {
+        result.push(currentValue);
+      }
+
+      currentValue = ATTENTION_SOUND_VOLUME - currentValue;
+    });
+
+    return Float32Array.from(result);
+  }
+
+  function ringtone() {
+    var audioChannel = 'notification';
+    var audioCtx = new AudioContext(audioChannel);
+
+    var o1 = audioCtx.createOscillator();
+    var o2 = audioCtx.createOscillator();
+    var gain = audioCtx.createGain();
+
+    var time = audioCtx.currentTime;
+    o1.type = o2.type = 'sine';
+    o1.frequency.value = 853;
+    o2.frequency.value = 960;
+
+    o1.start();
+    o2.start();
+    // Eventually stop the oscillator to allow garbage collecting.
+    o1.stop(time + 11);
+    o2.stop(time + 11);
+
+    var wave = getAttentionCurveWave();
+    gain.gain.setValueCurveAtTime(wave, time, 11);
+
+    o1.connect(gain);
+    o2.connect(gain);
+    gain.connect(audioCtx.destination);
```

Comment:
When the user presses "OK" the window is closed and thus this stops the alert too.